agent/provider/src/cli/rule.rs

```diff
@@ -82,22 +99,30 @@ fn set(set_rule: SetRule, config: ProviderConfig) -> Result<()> {
                 Some(_) => todo!("Setting rule for specific certificate isn't implemented yet"),
                 None => rules.set_default_audited_payload_mode(mode),
             },
-            SetOutboundRule::Partner(RuleWithCert::CertId { cert_id, mode }) => {
+            SetOutboundRule::Partner(PartnerRuleWithCert::CertId(CertId { cert_id, mode })) => {
                 rules.set_partner_mode(cert_id, mode)
             }
-            SetOutboundRule::Partner(RuleWithCert::ImportCert { import_cert, mode }) => {
+            SetOutboundRule::Partner(PartnerRuleWithCert::ImportCert { import_cert, mode }) => {
                 let mut keystore = CompositeKeystore::load(&rules.cert_dir)?;
 
-                let AddResponse { added, skipped } = keystore.add(&AddParams {
+                let AddResponse {
+                    invalid,
+                    leaf_cert_ids,
+                    ..
+                } = keystore.add_golem_cert(&AddParams {
                     certs: vec![import_cert],
                     permissions: vec![CertPermissions::All],
-                    whole_chain: true,
+                    whole_chain: false,
                 })?;
 
+                for cert_path in invalid {
+                    log::error!("Failed to import {cert_path:?}. Partner mode can be set only for Golem certificate.");
+                }
+
                 rules.keystore.reload(&rules.cert_dir)?;
 
-                for cert in added.into_iter().chain(skipped) {
-                    rules.set_partner_mode(cert.id(), mode.clone())?;
+                for cert_id in leaf_cert_ids {
+                    rules.set_partner_mode(cert_id, mode.clone())?;
```

Comment:
Keystore api seems that it doesn't fit here:
1. multiple leafs doesn't make any sense
2. returning `invalid` certs doesn't make any sense (if there was error when processing cert then why continue working in this case?)

But "milk has been spilled" already.